### PR TITLE
💄 Add the filename in the title of the main window

### PR DESCRIPTION
**Pull request by coliss86**
When opening a file, its filename is added to the title of the main window

image

### CODE REVIEW

**Comment by i-make-robots** [comment]
I would only display the last part of the filename.  firstly for length, and second to not reveal information in a screenshot that a user might forget about.

**Comment by coliss86** [comment]
I changed the title, I also put the basename before the name of the application like in inkscape

image

**Comment by i-make-robots** [comment]
Good. If we can capture that name in the right place it can be used for suggesting the save file name. The same mechanism would be used after a generator to set the generator as the suggested name.

File: src/main/java/com/marginallyclever/makelangelo/Makelangelo.java [hunk 1]
```diff
@@ -1054,6 +1058,15 @@ private void saveFile() {
 		}
 	}
 
+	private void setMainTitle(String filename) {
+		String title = "";
+		if (! "".equals(filename)) {
+			title += filename.replaceAll(".*/", "") + " - ";
```
Comment (PPAC37) [inline src/main/java/com/marginallyclever/makelangelo/Makelangelo.java:1064]: Windows Compatible '/' vs '\' ?

Comment (coliss86) [reply src/main/java/com/marginallyclever/makelangelo/Makelangelo.java]: Good catch, fixed

**Comment by PPAC37** [comment]
Perfect when opening a file.
There remains the case of I open a file then I use a generator ... I think that the file name will not be erased or updated but I have not tested.

**Comment by coliss86** [comment]
I implemented this: 
image

**Comment by PPAC37** [comment]
Simple and perfect! well played!

**Pull request merged**